src/common/utils.ts

```diff
@@ -5,31 +5,34 @@
 // that can be found in the LICENSE file exposed on Github (readium) in the project repository.
 // ==LICENSE-END==
 
-export function convertMultiLangStringToString(item: any): string {
-    if (typeof(item) === "string") {
-        return item;
-    }
-
-    // This is an object
-    const langs = Object.keys(item);
+import { Contributor } from "r2-shared-js/dist/es6-es2015/src/models/metadata-contributor";
+import { IStringMap } from "r2-shared-js/dist/es6-es2015/src/models/metadata-multilang";
+import { diMainGet } from "readium-desktop/main/di";
 
-    if (langs.length === 0) {
-        return null;
+// tslint:disable-next-line: max-line-length
+// https://readium2.herokuapp.com/pub/aHR0cHM6Ly9yYXcuZ2l0aHVidXNlcmNvbnRlbnQuY29tL0lEUEYvZXB1YjMtc2FtcGxlcy9tYXN0ZXIvMzAvcmVnaW1lLWFudGljYW5jZXItYXJhYmljL01FVEEtSU5GL2NvbnRhaW5lci54bWw%3D/manifest.json/show/all
+// tslint:disable-next-line: max-line-length
+// https://github.com/readium/webpub-manifest/blob/ff5c1e9e76ccc184d4d670179cfb70ced691fcec/schema/metadata.schema.json#L15-L32
+export function convertMultiLangStringToString(items: string | IStringMap | undefined): string {
```

Comment:
I would like to suggest the following changes in the code comments:
1) remove the `readium2.herokuapp.com` link, because this could change / become obsolete. Instead, copy/paste this concrete example (there is a bug in the parser, I will update this snippet later, when the fix is ready):
```
        "author": [
            {
                "name": {
                    "ar": "دافيد  خيّاط لبروفيسورا"
                }
            },
            {
                "name": {
                    "ar": "اردو هاتر ناتالي"
                }
            }
        ],
        "translator": {
            "name": {
                "ar": "فيّاض خليل مارينا"
            }
        },
        "contributor": {
            "name": "Vincent Gros",
            "role": "mrk"
        },
        "publisher": "Hachette Antoine",
```
...and provide a reference to the original EPUB:
https://github.com/IDPF/epub3-samples/blob/master/30/regime-anticancer-arabic/EPUB/package.opf

2) replace the JSON Schema reference with:
```
https://github.com/readium/webpub-manifest/blob/ff5c1e9e76ccc184d4d670179cfb70ced691fcec/schema/contributor-object.schema.json#L7-L24
https://github.com/readium/webpub-manifest/blob/ff5c1e9e76ccc184d4d670179cfb70ced691fcec/schema/metadata.schema.json#L15-L32
```

Reply:
I fixed a bug in the EPUB processor which was due to unclear parsing rules for default language (no explicit key in the object map).
So, this is the fixed version of the above JSON snippet (note the use of `_` object key to denote the default OPF language, but also note that instead of using `dc:language` (`ar`) which is already taken by a `refine`, we fallback to `_`):
```
        "author": [
            {
                "name": {
                    "ar": "دافيد  خيّاط لبروفيسورا",
                    "_": "Pr David Khayat"
                }
            },
            {
                "name": {
                    "ar": "اردو هاتر ناتالي",
                    "_": "Nathalie Hutter-Lardeau"
                }
            }
        ],
        "translator": {
            "name": {
                "ar": "فيّاض خليل مارينا",
                "_": "Marina Khalil Fayad"
            }
        },
        "contributor": {
            "name": "Vincent Gros",
            "sortAs": "Gros, Vincent",
            "role": "mrk"
        },
        "publisher": "Hachette Antoine",
```

Reply:
So, another round of fixes, now that `xml:lang` is handled properly in the OPF parser, resulting in:
```
        "author": [
            {
                "name": {
                    "ar": "دافيد  خيّاط لبروفيسورا",
                    "fr": "Pr David Khayat"
                }
            },
            {
                "name": {
                    "ar": "اردو هاتر ناتالي",
                    "fr": "Nathalie Hutter-Lardeau"
                }
            }
        ],
        "translator": {
            "name": {
                "ar": "فيّاض خليل مارينا",
                "fr": "Marina Khalil Fayad"
            }
        },
        "contributor": {
            "name": "Vincent Gros",
            "sortAs": "Gros, Vincent",
            "role": "mrk"
        },
        "publisher": "Hachette Antoine",
```

Reply:
done and fixes